hnn_core/feed.py

```diff
@@ -7,8 +7,47 @@
 import numpy as np
 
 
-class ExtFeed(object):
-    """The ExtFeed class of external spike input times.
+# based on cdf for exp wait time distribution from unif [0, 1)
+# returns in ms based on lamtha in Hz
+def _t_wait(prng, lamtha):
+    return -1000. * np.log(1. - prng.rand()) / lamtha
+
+
+def _get_prng(seed, gid, sync_evinput=False):
+    """Random generator for this instance.
+
+    Parameters
+    ----------
+    seed : int
+        The seed for random state generator.
+    gid : int
+        The cell ID
+    sync_evinput : bool
+        If True, all cells get the same prng
+
+    Returns
+    -------
+    prng : instance of RandomState
+        The seed for events assuming a given start time.
+    prng2 : instance of RandomState
+        The seed for start times.
+    """
+    # XXX: some param files use seed < 0 but numpy
+    # does not allow this.
+    if seed > 0:
+        prng2 = np.random.RandomState(seed)
+    else:
+        prng2 = None
+
+    if not sync_evinput:
+        seed = seed + gid
+
+    prng = np.random.RandomState(seed)
+    return prng, prng2
+
+
+def feed_event_times(feed_type, target_cell_type, params, gid):
```

Comment:
Would this be a good time to start explicitly defining the elements of the `params` argument for this function?

Reply:
not sure if it would be a good use of our time to document what is created as an output of `create_pext`. I think it _might_ be helpful to move `create_pext` into `feed.py` so folks can see what parameter corresponds to what. We would avoid duplicating what is in `params.py` just for documentation ... Let me know if you want me to do that or if you'd rather keep that in `params.py`

Reply:
I'm a strong +1 on leaving as-is for this PR. create_pext is a symptom of unwieldy parameter handling, so fixing (or documenting) one or two methods, I think, has lower priority than figuring out what to do about the root cause.

Reply:
We can wait on this. At some point we need to start dismantling the large param dicts that get passed around and I'm beginning to wonder if the best way to start this process is to begin explicitly defining function arguments where we can.

Reply:
Very likely so, let's see what we can do with a new feeds (drives) API

Reply:
+100 on explicitly defining function arguments where we can! It's the best way to avoid opaque code